src/crafting.cpp

```diff
@@ -767,24 +786,36 @@ const recipe *select_crafting_recipe( int &batch_size )
             if(display_mode >= 3 || display_mode <= 0) {
                 display_mode = 0;
             }
+        } else if (action == "CYCLE_NEXT_COMPONENT_LINE"){
+            if(component_scrolling_offset<0){
+                component_scrolling_offset = 0;
+            }else{
+                component_scrolling_offset = component_scrolling_offset + 1;
+            }
         } else if (action == "LEFT") {
             subtab = prev_craft_subcat( tab, subtab );
             redraw = true;
+            component_scrolling_offset = 0;
```

Comment:
Is there some way to do this that doesn't require every other command to handle this variable?  I can just see this getting missed when new commands are added and it going out of sync in weird ways.